src/lib.rs

```diff
@@ -528,6 +528,28 @@ impl<L, R> Either<L, R> {
         }
     }
 
+    /// Borrow the inner value as an iterator.
+    ///
+    /// ```
+    /// use either::*;
+    ///
+    /// let left: Either<_, &[u32]> = Left(vec![2, 3]);
+    /// let mut right: Either<Vec<u32>, _> = Right([4, 5].as_slice());
+    /// let mut all = vec![1];
+    /// all.extend(left.iter());
+    /// all.extend(right.iter());
+    /// assert_eq!(all, vec![1, 2, 3, 4, 5]);
+    /// ```
+    pub fn iter<'a>(
```

Comment:
If we're going to add `iter()`, we should probably also have `iter_mut()` as `self.as_mut().into_iter()`. I'm not sure these are terribly valuable, but there's precedent in the standard library.

Reply:
I added iter_mut as well.

I agree that wrapping up two chained method calls into one is not the pinacle of value, in fact the docs are probably more valuable than the code is, so if you don't feel like including `iter()` and `iter_mut()` perhaps I could add the expample to `into_inter()`?

Reply:
I think they're ok. They're simple, but as I mentioned `iter()` and `iter_mut()` are also in the standard library, like on `Option` where it is also equivalent to `.as_ref()` or `.as_mut()` first (but returning a different concrete type).